src/dune_pkg/mount.ml

```diff
@@ -50,7 +50,7 @@ let of_opam_url loc url =
          Path.relative dir file_digest
        in
        let+ path =
-         Tar.extract ~archive ~target
+         Archive_type.extract Archive_type.tar ~archive ~target
```

Comment:
```suggestion
         Archive_type.(extract tar ~archive ~target)
```
Proposition, but a matter of taste here.

Reply:
I tend to avoid this style because for example if the `Archive_type` module defined a value named `archive`, then it would be passed as the `~archive` argument rather than the local variable of the same name.